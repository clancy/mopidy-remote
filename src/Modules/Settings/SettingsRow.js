import React from 'react';
import { View, Text, StyleSheet, Image, TouchableOpacity } from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 12,
    flexDirection: 'row',
    alignItems: 'center',
  },
  text: {
    color: 'white',
    marginLeft: 12,
    fontSize: 16,
  }
});

const SettingsRow = (props) => (
  <View style={styles.container}>
    <TouchableOpacity
      onPress={props.onPress} >
      <Text style={styles.text}>
        {props.text}
      </Text>
    </TouchableOpacity>

  </View>
);

export default SettingsRow;
