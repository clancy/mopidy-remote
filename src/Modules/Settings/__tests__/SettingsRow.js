import 'react-native';
import React from 'react';
import SettingsRow from '../SettingsRow';
import { shallow, mount, render } from 'enzyme';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
describe('SettingsRow tests', () => {
  test('renders SettingsRow correctly', () => {
    const tree = renderer.create(
      <SettingsRow
        text={"Text"}
        onPress={() => {}}
        />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('SettingsRow onPress calls passed function', () => {
    var controlMock = jest.fn();
    const tree = shallow(
      <SettingsRow
        text={"Text"}
        onPress={() => { controlMock('onPress') }}
        />
    );

    tree.find('TouchableOpacity').simulate('press');

    expect(controlMock.mock.calls.length).toBe(1);
    expect(controlMock.mock.calls[0][0]).toBe('onPress');
  });
});
