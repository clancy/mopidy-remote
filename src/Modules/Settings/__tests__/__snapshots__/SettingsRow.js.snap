// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsRow tests renders SettingsRow correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "padding": 12,
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontSize": 16,
          "marginLeft": 12,
        }
      }
    >
      Text
    </Text>
  </View>
</View>
`;
